dev/tasks/r/github.packages.yml

```diff
@@ -375,29 +375,30 @@ jobs:
           read_parquet(system.file("v0.7.1.parquet", package = "arrow"))
           print(arrow_info())
 
-  #TODO test macos source build?
-  test-linux-source:
+  test-source:
     needs: source
-    name: Test linux source build
-    runs-on: ubuntu-latest
+    name: Test {{ '${{ matrix.platform.name }}' }} source build
+    runs-on: {{ '${{ matrix.platform.runs_on }}' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - {runs_on: "ubuntu-latest", name: "Linux"}
+          - {runs_on: ["self-hosted", "macos-10.13"] , name: "macOS"}
     steps:
       - name: Install R
+        if: matrix.platform.name == 'Linux'
         uses: r-lib/actions/setup-r@v2
-        with:
-          install-r: false
       {{ macros.github_setup_local_r_repo(false, false)|indent }}
-      {{ macros.github_checkout_arrow()|indent }}
+      {{ macros.github_checkout_arrow(action_v="3")|indent }}

```

Comment:
It seems that we don't need this for macOS.

Reply:
We do need this for the 10.13 runner, v4 uses node 20 which errors on 10.13

Reply:
Ah, I wanted to say that we don't need to checkout apache/arrow on macOS.

Reply:
Oh I see what you mean...  I can't think of an elgant fix at the moment feel free to push something :D

Reply:
Could you make it a separate build step and use an `if:`?